### ISSUE TITLE
Transfer ownership to nponeccop/libdanga-socket-perl

### ISSUE THREAD

**Issue by nponeccop** [nponeccop/libdanga-socket-perl#2]
Ok now that I've got co-maintainer rights from Brad and have released https://metacpan.org/release/NML/Danga-Socket-1.62 it's time to make my github repo the official one. 

Unfortunately there is confusion as I forked my repo from yours without knowing it's unofficial. So I don't know what to do so my repo appears as the top one in the network graph.

A simple solution would be just to delete this repo and re-fork it from mine. Or use Transfer Ownership. Or we could leave it as is and I'd just enable issues in my own repo. Note that to use "transfer ownership" I need to delete mine first. 

Or we might leave it as is unless some _real_ confusion occurs, as opposed to hypotetic one.

**Comment by apparentlymart** [comment]
If you'd be okay with your repository being unavailable briefly for a few minutes then I'd propose the following:
* Delete your existing fork.
* Create a new, empty repository called `libdanga-socket-perl` in your account, and use local `git` to pull `master` from this repository and push it to your repository, so GitHub is none the wiser that the commits originated here.
* I'll delete or archive this repository once you're done, to make it extra clear that there's no further development going on here. I don't generally write Perl these days, so I don't have any immediate need for this repository, but I'd lean towards just archiving it at first in case others have cloned from it or are otherwise expecting it to be here.

I could potentially delete this repository now, but I'm not sure what effect that would have on existing forks like yours, and has the risk of causing breakage/confusion for anyone that was already relying on this repository in some way. While I might still delete this repository _eventually_, I'd rather let it sit in the archived state for a while first to give people some time to notice and move away.

How does that sound? Is briefly having your repository deleted acceptable to you in order to break the link?

**Comment by nponeccop** [comment]
I think that the Transfer Ownership feature is designed specifically for this scenario. You deletes the repo and Github redirects any requests to it to my repo instead. So the people who might have your repo cloned or bookmarked won't get lost:

https://help.github.com/en/github/administering-a-repository/transferring-a-repository

I let you decide though. I'm pretty ready to proceed with your scenario instead of transfering.

Let me know and I'll proceed. I apparently need to delete my repo on GitHub in both your scenario and in case of transfering, but I won't do so until we decide.

**Comment by apparentlymart** [comment]
I'm happy to transfer ownership of this repository to you. I believe that will also require your repository to be briefly deleted in order to make room for the incoming repository, but perhaps it's possible for you to do that only after I've initiated the transfer request in order to minimize the time the repository does not exist in your account.

I would note that I'm not sure that my repository here is totally up-to-date with what's released on CPAN. After doing a little detective work I see the branch `anyevent` which makes me think I'd originally created this repository to work on [Danga::Socket::AnyEvent](https://metacpan.org/pod/Danga::Socket::AnyEvent), which I apparently released for the first time in 2012 if we ignore the initial dev-only release from 2009. That suggests that this repository likely forked from the upstream Subversion repository (formerly hosted by Six Apart, IIRC) at that time, which seems promising for it to be at least new enough to include the code from the last BRADFITZ release in 2008. So :crossed_fingers: it's up-to-date.

I'm honestly not sure I still know my PAUSE login at this point because I've been away from Perl for many years, but if you'd be interested I'm happy to try to give you release rights on `Danga::Socket::AnyEvent` too, in case that's useful in some way, since I believe I'd be transferring to you the primary repository for that fork as well. If that's not something you're interested in, no worries... `Danga::Socket::AnyEvent` was briefly in use inside Say Media but I'd be surprised if it has many current users.

**Comment by apparentlymart** [comment]
Actually, I just noticed that I also have [libdanga-socket-anyevent-perl](https://github.com/apparentlymart/libdanga-socket-anyevent-perl), so I assume the `anyevent` branch in this repository was just from my preparation to fork, not where I made the releases from. Sorry for the diversion.

**Comment by nponeccop** [comment]
> I'm honestly not sure I still know my PAUSE login at this point

Apparently your CPAN login is MART: https://metacpan.org/author/MART

> If that's not something you're interested in, no worries... 

I only maintain `Danga::Socket` because it's a dependency of `Gearman-Server` which is a test dependency of `AnyEvent-Gearman` which I use in production.

So unfortunately I don't want to maintain Danga::Socket::AnyEvent. There is a lot of maintenance work even with what I have: I managed to have forks of 4 packages: http://fast-matrix.cpantesters.org/?author=NML, they were all broken. For AnyEvent::Gearman there are like 2 dozen of changes: https://github.com/typester/anyevent-gearman-perl/pull/18. And there are more packages in optional dependencies of optional dependencies which I still need to fix.

> I'm not sure that my repository here is totally up-to-date with what's released on CPAN

I checked the repo against the last tarball on CPAN of course, and it appears up to date.
